src/DIRAC/FrameworkSystem/Service/ProxyManagerHandler.py

```diff
@@ -411,39 +413,32 @@ def export_getVOMSProxyWithToken(self, userDN, userGroup, requestPem, requiredLi
 
     def export_exchangeProxyForToken(self):
         """Exchange a proxy for an equivalent token to be used with diracx"""
-        try:
-            from diracx.routers.auth import (  # pylint: disable=import-error
-                AuthSettings,
-                create_token,
-                TokenResponse,
-            )  # pylint: disable=import-error
-
-            authSettings = AuthSettings()
-
-            from uuid import uuid4
-
-            credDict = self.getRemoteCredentials()
-            vo = Registry.getVOForGroup(credDict["group"])
-            payload = {
-                "sub": f"{vo}:{credDict['username']}",
-                "vo": vo,
-                "aud": authSettings.token_audience,
-                "iss": authSettings.token_issuer,
-                "dirac_properties": list(
-                    set(credDict.get("groupProperties", [])) | set(credDict.get("properties", []))
-                ),
-                "jti": str(uuid4()),
+
+        from diracx.core.preferences import get_diracx_preferences
+
+        apiKey = gConfig.getValue("/DiracX/LegacyExchangeApiKey")
+        if not apiKey:
+            return S_ERROR("Missing mandatory /DiracX/LegacyExchangeApiKey configuration")
+
+        diracxUrl = get_diracx_preferences().url
```

Comment:
Maybe this should be something like:
```suggestion
        diracxUrl = gConfig.getValue("/DiracX/URL")
        if not diracxUrl:
            return S_ERROR("Missing mandatory /DiracX/URL configuration")
```

Reply:
:+1: